### PR TITLE
Resolves #4559: Changes land rights buttons to be disabled if no tiles for sale remain

### LINKED ISSUE #4559: Feature request: Disable Buy Land / Construction Rights if there's nothing to buy
Some scenarios don't have any land or construction rights for sale. And all of them only have a finite amount of rights for sale. So when there's nothing left to buy, it would be cool to give that information to the user in some way. I think this would be the simplest way to do that:

Mockup for when there's no land or construction rights for sale:

image

(Don't even let them click the button)

Mockup for when there's one type of land right for sale, but not the other (forces toggle to the remaining type)

image

I tried to figure out the source code that governs the land purchasing but I only got as far as finding OWNERSHIP_AVAILABLE.

https://github.com/OpenRCT2/OpenRCT2/blob/43577eb7b946e382ab33640307def08f6e13873b/src/world/park.c#L961

### PR DESCRIPTION
**Pull request by Overv**
This commit changes the land rights button in the park window to be
disabled if there are no more land rights and construction rights for
sale. It also disables the individual land rights and construction
rights buttons in the land rights window itself if that specific type of
tile is no longer for sale.

This is implemented by introducing two new global variables called
gLandRemainingOwnershipSales and gLandRemainingConstructionSales. These
are updated by map_count_remaining_land_rights() to contain the number
of land rights and construction rights that remain for sale. This
function is called when a scenario is loaded and whenever the game
command to buy land rights or construction rights is invoked.

It also introduces three new tooltip strings that explain why the
buttons are disabled.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
I think that would work better, yes.

**Comment by Overv** [comment]
@Gymnasiast @RollingStar I've changed the string.

File: src/world/map.c [hunk 1]
```diff
@@ -395,6 +398,30 @@ void map_init(int size)
 }
 
 /**
+ * Counts the number of surface tiles that offer land ownership rights for sale,
+ * but haven't been bought yet. It updates gLandRemainingOwnershipSales and
+ * gLandRemainingConstructionSales.
+*/
+void map_count_remaining_land_rights() {
```
Comment (Gymnasiast) [inline src/world/map.c]: This left brace should go onto its own line.

**Comment by Gymnasiast** [comment]
Good, I left a small line note about formatting, but other than that it looks good.

**Comment by fromthere** [comment]
If i use sandbox mode to add some land to purchase or land to buy build rights, disable buttons are not enabled. also, there's no way check where i bought build right since if there's no more build right for sale

**Comment by Overv** [comment]
@fromthere Can you create an issue? The sandbox mode problem is easy to fix, but we'll have to discuss the other one.

**Pull request merged**